src/nvim/cursor_shape.c

```diff
@@ -85,18 +85,18 @@ char_u *parse_shape_opt(int what)
             len = 1;
           else
             len = 2;
-          if (len == 1 && TOLOWER_ASC(modep[0]) == 'a')
+
+          if (len == 1 && TOLOWER_ASC(modep[0]) == 'a') {
             all_idx = SHAPE_IDX_COUNT - 1;
-          else {
+          } else {
             for (idx = 0; idx < SHAPE_IDX_COUNT; ++idx)
-              if (STRNICMP(modep, shape_table[idx].name, len)
-                  == 0)
+              if (STRNICMP(modep, shape_table[idx].name, len) == 0)
                 break;
-            if (idx == SHAPE_IDX_COUNT
-                || (shape_table[idx].used_for & what) == 0)
+          
+	    if (idx == SHAPE_IDX_COUNT || (shape_table[idx].used_for & what) == 0)
               return (char_u *)N_("E546: Illegal mode");
-            if (len == 2 && modep[0] == 'v' && modep[1] == 'e')
-              found_ve = TRUE;
+	    if ((len = 2 && modep[0] == 'v' && modep[1] == 'e'))
```

Comment:
The extra parens weren't from the lint; they were from GCC actually. The compiler told me that the statement was ambiguous.